corfudb-tools/src/main/java/org/corfudb/compactor/CorfuStoreCompactorMain.java

```diff
@@ -0,0 +1,336 @@
+                                        .newBuilder().setFileDescriptor(fileDescriptorProto).build();
+                                CorfuRecord<ProtobufFileDescriptor, TableMetadata> corfuRecord =
+                                        descriptorTable.putIfAbsent(fileName, new CorfuRecord<>(fileDescriptor, null));
+                                if (corfuRecord == null) {
+                                    log.info("Add proto file {}, fileDescriptor {} to ProtobufDescriptorTable",
+                                            fileName, fileDescriptor.getFileDescriptor());
+                                }
+                                // construct a new tableDescriptorsMap using default FileDescriptorProto instances
+                                tableDescriptorsBuilder.putFileDescriptors(protoName,
+                                        fileDescriptorProto.getDefaultInstanceForType());
+                            });
+
+                    tableDescriptorsBuilder.setKey(registryRecord.getPayload().getKey());
+                    tableDescriptorsBuilder.setValue(registryRecord.getPayload().getValue());
+                    tableDescriptorsBuilder.setMetadata(registryRecord.getPayload().getMetadata());
+
+                    // clean up FileDescriptorsMap inside RegistryTable to optimize memory
+                    TableDescriptors tableDescriptors = tableDescriptorsBuilder.build();
+                    registryTable.put(tableName, new CorfuRecord<>(tableDescriptors, registryRecord.getMetadata()));
+
+                    log.info("Cleaned up an entry in RegistryTable: {}${}",
+                            tableName.getNamespace(), tableName.getTableName());
+                }
+                tx.commit();
+                log.info("syncProtobufDescriptorTable: completed!");
+                break;
+            } catch (TransactionAbortedException txAbort) {
+                if (numRetries-- <= 0) {
+                    throw txAbort;
+                }
+                log.info("syncProtobufDescriptorTable: commit failed. Will retry {} times. Cause {}",
+                        numRetries, txAbort);
+            }
+        }
+    }
+
+    private static void getCompactorArgs(String[] args) {
+        // Parse the options given, using docopt.
+        Map<String, Object> opts =
+                new Docopt(USAGE)
+                        .withVersion(GitRepositoryState.getRepositoryState().describe)
+                        .parse(args);
+        hostname.add(opts.get("--hostname").toString());
+        port = Integer.parseInt(opts.get("--port").toString());
+
+        if (opts.get("--keystore") != null) {
+            runtimeKeyStore = opts.get("--keystore").toString();
+        }
+        if (opts.get("--ks_password") != null) {
+            runtimeKeystorePasswordFile = opts.get("--ks_password").toString();
+        }
+        if (opts.get("--truststore") != null) {
```

Comment:
these code also copy pasted from corfu, could you please reuse what we have in corfu